### DISCUSSION TITLE
Partials with inline templates

### DISCUSSION THREAD

**Discussion by dsyer** [jstachio/jstachio#277] [Q&A] [answered]
I thought maybe I could write `{{>foo}}` in a template and have it render from `@JStache(path="foo", template="<span>bar</span>")` but that seems to be illegal (compiler fails because "foo.mustache" is not found)? Am I missing something?

**Comment by agentgt** [comment]
Can you post a little more code?

You can’t combine template and path arguments. I think path takes precedence.

**Reply by dsyer** [reply]
There isn't really any more code. The path does take precedence on compilation so that's a problem, because it can't take a hint and resolve the template from the annotation. Seems like a bug to me, but I'm open to suggestions.

**Reply by agentgt** [reply]
I think you want JStachePartial.

**Reply by agentgt** [reply]
If you want an inline template as a partial that is what JStachePartials does.

**Reply by dsyer** [reply, marked as answer]
I get it. I never think of putting annotations at package level, but that's probably what I want here.:

```java
@JStachePartials(@JStachePartial(name = "foo", template = "<span>bar</span>"))
package com.example.test;
```

What about something that is shared between packages (and don't say "annotate the module")?

**Reply by dsyer** [reply]
I got it. Use `@JStacheConfig(using = DemoApplication.class)` at the package level,

**Reply by agentgt** [reply]
>  (and don't say "annotate the module")

ha 😆 

> I got it. Use @JStacheConfig(using = DemoApplication.class) at the package level,

A feature that you helped with! I'm glad you were persistent on the module pain.

Overall I think a general problem with JStachio doc is I went a little too hard on the "wikipedia" style instead of complete single reference doc style. The overview: https://jstach.io/jstachio/ needs to have the code examples copied that are sprinkled through the classes javadoc. Basically everything that is on https://jstach.io/jstachio/io.jstach.jstache/io/jstach/jstache/JStache.html needs to be sort of copied into the overview.html (https://jstach.io/jstachio/).

**Reply by dsyer** [reply]
> The overview: https://jstach.io/jstachio/ needs to have the code examples copied that are sprinkled through the classes javadoc.

+1 to that. I have been through that process quite a few times.
